src/services/AmazonS3Service.ts

```diff
@@ -0,0 +1,74 @@
+import axios, { AxiosError, AxiosResponse } from 'axios';
+
+import { Image } from '../types/image';
+
+const maxImageSize = 1000000;
+
+class AmazonS3Service {
+  onFileChange(e: React.ChangeEvent<HTMLInputElement> | DragEvent): Promise<Image> {
+    // Get the file from selection or drag and drop
+    const files =
+      (e as React.ChangeEvent<HTMLInputElement>)?.target?.files ||
+      (e as DragEvent)?.dataTransfer?.files ||
+      [];
+    if (!files.length) {
+      return Promise.resolve({ imageBinary: '', imageType: '' });
+    }
+    const file = files[0];
+    // Convert file into binary string
+    const reader = new FileReader();
+    return new Promise((resolve, reject) => {
+      reader.onload = (e: ProgressEvent<FileReader>) => {
+        // Only allow jpg, jpeg, png, and gifs to be selected
+        const file = e?.target?.result as string;
+        const fileRegexMatches = file.match(/image\/(jpe?g|png|gif)/);
+        if (fileRegexMatches === null) {
+          return reject(new Error('Wrong file type. Only images are allowed.'));
+        }
+        // Restrict the size of the file
+        if (file.length > maxImageSize) {
+          return reject(new Error('Image is loo large.'));
+        }
+        return resolve({
+          imageBinary: file,
+          imageType: fileRegexMatches[1],
+        });
+      };
+      reader.readAsDataURL(file);
+    });
+  }
+
+  uploadImage(image: Image, promotionId: string) {
+    // Get the presigned URL
+    axios
+      .get(process.env.REACT_APP_AWS_PRESIGNED_URL_ENDPOINT as string, {
+        params: {
+          promotionId,
+          imageType: image.imageType,
+        },
+      })
+      .then((response: AxiosResponse) => {
+        // Upload the image
+        const binary = atob(image.imageBinary.split(',')[1]);
+        const array = [];
+        for (let i = 0; i < binary.length; i++) {
+          array.push(binary.charCodeAt(i));
+        }
+        const blobData = new Blob([new Uint8Array(array)], {
+          type: 'image/' + image.imageType,
+        });
+        const result = fetch(response.data.uploadURL, {
```

Comment:
For this, are you able to use axios instead of fetch? I was thinking because you used axios on line 43 we could just keep using that

Reply:
I don't think axios allows uploading blob, so we might have to stick with fetch

Reply:
Oh interesting, sounds good then we can keep this!